lua/telescope/previewers/init.lua

```diff
@@ -1,36 +1,280 @@
+---     - `require('telescope.previewers.utils').ts_highlighter(bufnr, ft)`
+---   - If you want to use `vim.fn.search` or similar you need to run it in
+---     that specific buffer context. Do
+--- <pre>
+---       vim.api.nvim_buf_call(bufnr, function()
+---         -- for example `search` and `matchadd`
+---       end)
+---     to achieve that.
+--- </pre>
+---   - If you want to read a file into the buffer it's best to use
+---     `buffer_previewer_maker`. But access this function with
+---     `require('telescope.config').values.buffer_previewer_maker`
+---     because it can be redefined by users.
+previewers.new_buffer_previewer = buffer_previewer.new_buffer_previewer
 
-previewers.new_buffer_previewer   = buffer_previewer.new_buffer_previewer
+--- A universal way of reading a file into a buffer previewer.
+--- It handles async reading, cache, highlighting, displaying directories
+--- and provides a callback which can be used, to jump to a line in the buffer.
+---@param filepath string: String to the filepath, will be expanded
+---@param bufnr number: Where the content will be written
+---@param opts table: keys: `use_ft_detect`, `bufname` and `callback`
 previewers.buffer_previewer_maker = buffer_previewer.file_maker
-previewers.vim_buffer_cat         = buffer_previewer.cat
-previewers.vim_buffer_vimgrep     = buffer_previewer.vimgrep
-previewers.vim_buffer_qflist      = buffer_previewer.qflist
-previewers.git_branch_log         = buffer_previewer.git_branch_log
-previewers.git_commit_diff        = buffer_previewer.git_commit_diff
-previewers.git_file_diff          = buffer_previewer.git_file_diff
-previewers.ctags                  = buffer_previewer.ctags
-previewers.builtin                = buffer_previewer.builtin
-previewers.help                   = buffer_previewer.help
-previewers.man                    = buffer_previewer.man
-previewers.autocommands           = buffer_previewer.autocommands
-previewers.highlights             = buffer_previewer.highlights
-previewers.display_content        = buffer_previewer.display_content
 
-previewers.Previewer = Previewer
+
+--- A previewer that is used to display a file. It uses the `buffer_previewer`
+--- interface and won't jump to the line. To integrate this one into your
+--- own picker make sure that the field `path` or `filename` is set for
+--- each entry.
+--- The preferred way of using this previewer is like this
+--- `require('telescope.config').values.file_previewer`
+--- This will respect user configuration and will use `termopen_previewer` in
+--- case it's configured that way.
+previewers.vim_buffer_cat = buffer_previewer.cat
+
+--- A previewer that is used to display a file and jump to the provided line.
+--- It uses the `buffer_previewer` interface. To integrate this one into your
+--- own picker make sure that the field `path` or `filename` and lnum is set
+--- in each entry. If the latter is not present it will default to the first
+--- line.
```

Comment:
```suggestion
--- It uses the `buffer_previewer` interface. To integrate this one into your
--- own picker make sure that the field `path` or `filename` and `lnum` is set
--- in each entry. If the latter is not present, it will default to the first
--- line.
```